### PR TITLE
strace: update to 5.10.

### PR DESCRIPTION
**Pull request by olafmersmann**

### CODE REVIEW

File: srcpkgs/strace/template [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
+changelog="https://raw.githubusercontent.com/strace/strace/master/NEWS"
 distfiles="https://github.com/strace/strace/releases/download/v${version}/strace-${version}.tar.xz"
-checksum=39473eb8465546c3e940fb663cb381eba5613160c7302794699d194a4d5d66d9
+checksum=fe3982ea4cd9aeb3b4ba35f6279f0b577a37175d3282be24b9a5537b56b8f01c
 
 case "$XBPS_TARGET_MACHINE" in
 	aarch64*|ppc64*) configure_args+=" --enable-mpers=no" ;;
 	# work around https://github.com/strace/strace/issues/133
 	x86_64-musl) configure_args+=" --enable-mpers=no" ;;
 esac
```
Comment (ericonr) [inline srcpkgs/strace/template:20]: Given https://github.com/strace/strace/commit/1b57b2241aef3e3ee5ea4cd60772252b15254de8 , can you try to remove this block?

File: srcpkgs/strace/template [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
+
+do_check() {
+	# XXX: Don't run the complete test suite because it takes
+	# about ~90 minutes and some tests fail in CI.
+	make check TESTS="bpf.gen epoll_pwait.gen getcpu.gen open.gen read-write.gen readlink.gen seccomp-filter.gen mmap ioctl caps readv"
```
Comment (ericonr) [inline srcpkgs/strace/template:25]:
```suggestion
	make check \
		TESTS="bpf.gen epoll_pwait.gen getcpu.gen open.gen read-write.gen readlink.gen seccomp-filter.gen mmap ioctl caps readv"
```

Would be nice if there was a target to explicitly exclude some tests.

File: srcpkgs/strace/template (hunk 1 above)
Comment (olafmersmann) [reply srcpkgs/strace/template:20]: I tried but configure fails on both platforms with
```
checking whether to enable m32 personality support... no
configure: error: Cannot enable m32 personality support
```

Comment (ericonr) [reply srcpkgs/strace/template:20]: Ah, shame :/

**Comment by ericonr** [comment]
@olafmersmann could you take a look at the manual? I added a `full` option for tests (`$XBPS_CHECK_PKGS = full`), maybe we could run all the tests if that's set?

**Comment by olafmersmann** [comment]
> @olafmersmann could you take a look at the manual? I added a `full` option for tests (`$XBPS_CHECK_PKGS = full`), maybe we could run all the tests if that's set?

If XBPS_CHECK_PKGS is set to full, do_check() runs all tests, otherwise only a small (arbitrary) selection of tests that should not fail in CI.

**Comment by ericonr** [comment]
Thank you, wonderful :)

**Pull request merged**